scripts/goldpolish-target-extract-seq.py

```diff
@@ -0,0 +1,178 @@
+    )
+    parser.add_argument(
+        "-b",
+        "--bed",
+        help="bed file specifying regions to polish",
+        type=str,
+        required=False,
+    )
+    parser.add_argument(
+        "-o",
+        "--output",
+        help="prefix of output file [<output>.fa]",
+        type=str,
+        required=False,
+        default="GoldPolish-Target_extracted_gaps.fa",
+    )
+    parser.add_argument(
+        "-l",
+        "--length",
+        help="length of flanking regions [64]",
+        type=int,
+        required=False,
+        default=64,
+    )
+    return parser.parse_args()
+
+
+def make_coord_dict(bed):
+    """Creates a dictionary of tuples representing regions to polish"""
+    coord_dict_2 = {}
+
+    with open(bed, encoding="utf-8") as bed_file:
+        bed_reader = csv.reader(bed_file, delimiter="\t", quotechar='"')
+        for row in bed_reader:
+            contig_name = row[0]
+            coord = Coordinate(row[1], row[2])
+
+            if contig_name not in coord_dict_2:
+                coord_dict_2[contig_name] = [coord]
+            else:
+                coord_dict_2[contig_name].append(coord)
+    return coord_dict_2
+
+
+def extract_masked_subsequences(sequence, name, flank_length, writer):
+    """extracts lowercase sequences and returns list of sequences with flank_length bp long flanking regions"""
+    gap_count = 1
+    idx = 1  # index of subseqs
+
+    subseqs = re.findall(r"([A-Z]+|[a-z]+)", sequence)
+    filtered_subseqs = [] # short uppercase seqs are appended to adjacent softmasked seqs
+    filtered_subseqs.append(
+        subseqs[0]
+    )  # first item always appended, avoids 0-index issues
+
+    while idx < len(subseqs):
+        subseq = subseqs[idx]
+        if subseq.isupper():
+            if len(subseq) >= 2 * flank_length:  # surpases min length threshold
+                filtered_subseqs.append(subseq)
+            elif filtered_subseqs[-1].islower():
+                # append lower case subseq to previous subseq
+                filtered_subseqs[-1] = filtered_subseqs[-1] + subseq.lower()
+            else:
+                raise Exception("Unexpected order of softmasked subseqs")
+        else:  # subseq is lower
+            if filtered_subseqs[-1].isupper():
+                filtered_subseqs.append(subseq)
+            elif filtered_subseqs[-1].islower():
+                # previous subseq is lower, need to concat with prev
+                filtered_subseqs[-1] = filtered_subseqs[-1] + subseq
+            else:
+                raise Exception("Unexpected order of softmasked subseqs")
+        idx += 1
+        
+    idx = 0
+
+    for subseq in filtered_subseqs:
+        if subseq.islower() and len(subseq) > min_gap_length:
+            flank_start = max(0, idx - flank_length)
+            flank_end = min(len(sequence), idx + len(subseq) + flank_length)
+            if flank_end >= flank_start:  # just in case
```

Comment:
Wasn't sure what the resolution was here? Wouldn't it being an empty sequence be a reason to also skip writing it?